app/src/main/java/sdp/moneyrun/MenuActivity.java

```diff
@@ -219,17 +215,27 @@ private void setNavigationViewListener() {
     }
 
     private void setPutExtraArguments(Intent intent){
-
-        intent.putExtra("playerId",playerId);
-        intent.putExtra("playerId"+playerId,playerInfo);
+        intent.putExtra("user", user);
     }
 
-    public void onButtonSwitchToUserProfileActivity(View view) {
-
+    public void onButtonSwitchToUserProfileActivity() {
         Intent playerProfileIntent = new Intent(MenuActivity.this, PlayerProfileActivity.class);
         setPutExtraArguments(playerProfileIntent);
         startActivity(playerProfileIntent);
     }
+
+    public void onButtonSwitchToLeaderboardActivity() {
+        Intent leaderboardActivity = new Intent(MenuActivity.this, LeaderboardActivity.class);
+        setPutExtraArguments(leaderboardActivity);
+        startActivity(leaderboardActivity);
+    }
+
+    public void onButtonSwitchToLoginActivity(){
```

Comment:
I think when going back to the login activity (Logging out), the current activity should be closed (ie. finish())

Reply:
indeed, done